### PR TITLE
feat: robotoff anonymous questions

### PR DESCRIPTION
**Pull request by alexgarel**
### What

feat: Make it possible for anonymous users to see and answer robotoff questions
build: some changes to be compatible with a local robotoff instance

fixes: 
- #3158

### CODE REVIEW

File: templates/web/pages/product/product_page.tt.html [hunk 1]
```diff
@@ -47,7 +47,7 @@
     <p>→ <a href="[% public_product_url %]">[% lang("product_page_on_the_public_database") %]</a></p>
 [% END %]
 
-[% IF (user_id.defined) && (robotoff_url.defined) && (robotoff_url.length > 0) %]
+[% IF (robotoff_url.defined) && (robotoff_url.length > 0) %]
```
Comment (alexgarel) [inline templates/web/pages/product/product_page.tt.html:50]: yes it's that easy, because robotoff takes care of computing device_id using url.

Comment (raphael0202) [inline templates/web/pages/product/product_page.tt.html:50]: Maybe it could it a good idea to add a comment in the code to explain this

Comment (stephanegigandet) [reply templates/web/pages/product/product_page.tt.html:50]: so for a non logged in OFF website user, how is the device_id computed? is it a hash of the user ip address or something?

Comment (alexgarel) [reply templates/web/pages/product/product_page.tt.html:50]: yes it's the sha1 of user ip, see https://github.com/openfoodfacts/robotoff/blob/e46942f0055b54183ef5570b7c39215ed312309c/robotoff/app/api.py#L235

Comment (alexgarel) [reply templates/web/pages/product/product_page.tt.html:50]: done.

**Review by raphael0202 — APPROVED** [review]

**Pull request merged**